schema.json

```diff
@@ -1,255 +1,266 @@
-            }
-          },
-          "required": ["name", "fields", "type"]
+        "unit": {
+          "description": "The unit type, defined in Table A-13 of E1.20.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 255
         }
-      ]
+      },
+      "required": [ "type" ]
+    },
+    "labeledIntegerType": {
+      "description": "Associates a name to an integer value.",
+      "type": "object",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "value": { "type": "integer" }
+      },
+      "required": [ "value" ]
     },
-    "label": {
+    "listType": {
+      "description": "A list of objects all having the same type.",
       "type": "object",
       "properties": {
-        "label": {
-          "minLength": 1,
-          "type": "string"
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": { "const": "list" },
+        "itemType": {
+          "$ref": "#/definitions/oneOfTypes"
         },
-        "value": {
+        "minItems": {
+          "description": "The minimum list size.",
+          "type": "integer",
+          "minimum": 0
+        },
+        "maxItems": {
+          "description": "The maximum list size.",
           "type": "integer",
-          "maximum": 4294967295,
           "minimum": 0
         }
       },
-      "required": ["label", "value"]
+      "required": [ "type", "itemType" ]
     },
-    "name": {
+    "nameType": {
+      "description": "A name is a string having a length of at least 1.",
       "type": "string",
       "minLength": 1
     },
-    "range": {
+    "oneOfTypes": {
+      "description": "One of any of the types. This provides a single location
+                      to keep the list.",
+      "oneOf": [
+        { "$ref": "#/definitions/bitFieldType" },
+        { "$ref": "#/definitions/compoundType" },
+        { "$ref": "#/definitions/integerType" },
+        { "$ref": "#/definitions/listType" },
+        { "$ref": "#/definitions/pdEnvelopeType" },
+        { "$ref": "#/definitions/simpleType" },
+        { "$ref": "#/definitions/stringType" },
+        { "$ref": "#/definitions/subdeviceType" }
+      ]
+    },
+    "subdeviceType": {
+      "description": "A subdevice type. One of the root device (e.g. value 0),
+                      the root or all subdevices or the broadcast subdevice,
+                      the root or any subdevice, or just any subdevice.",
+      "type" : "string",
+      "enum": [
+            "root",
+            "root_or_any_subdevice_or_broadcast",
+            "root_or_any_subdevice",
+            "any_subdevice"
+          ]
+    },
+    "pdEnvelopeType": {
+      "description": "Contains a length/data pair for one Parameter Data item,
+                      where the length is an unsigned 8-bit value and the data
+                      has 'length' bytes. This exists to provide a schema
+                      definition for the 'envelope' of a PDL/PD pair.",
```

Comment:
`PDL/PD` should be `PD/PDL` given the order they come in the packet.

Reply:
It's common for structures to be specified as "length", "data". Also, "PDL" comes before "PD" in the packet. Are you thinking of "PID"?